src/App.js

```diff
@@ -4,9 +4,31 @@ import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export class App extends Component {
-  state = {};
+  state = {
+    search: '',
+    movies: moviesFromServer,
+  };
+
+  handleSearchInput = (event) => {
+    this.setState({
+      search: event.target.value,
+    }, this.moviesFilter);
+  }
+
+  moviesFilter = () => {
+    this.setState(prevState => ({
+      ...prevState,
+      movies: moviesFromServer
+        .filter(movie => movie.title
+          .toLowerCase().includes(prevState.search.toLowerCase())
+          || movie.description
+            .toLowerCase().includes(prevState.search.toLowerCase())),
+    }));
+  }
```

Comment:
```suggestion
  moviesFilter = () => {
    this.setState({
      movies: moviesFromServer
        .filter(movie => movie.title
          .toLowerCase().includes(prevState.search.toLowerCase())
          || movie.description
            .toLowerCase().includes(prevState.search.toLowerCase())),
    });
  }
```

You don't need the previous state here. You changing only 'movies' field